### PR TITLE
WAS-1013 reviewing exemptions back-link fix

### PR DESCRIPTION
**Pull request by rickclare**

### CODE REVIEW

**Comment by rickclare** [comment]
@woodpigeon Would you mind reviewing this as the original author of this `BackLinkHelper`

File: app/helpers/digital_services_core/back_link_helper.rb [hunk 1]
```diff
@@ -4,27 +4,41 @@ module BackLinkHelper
...
-      def tag
-        return content_tag(:br) if enrollment_is_already_complete?
-        title, url = link_arguments(enrollment)
-        link_to title, url, class: "back-link"
+      def tag(text = nil, html_opts = {})
+        if enrollment && (current_request.path == enrollment_reviewing_path ||
+                          enrollment_is_already_complete?)
+          content_tag(:br)
+        else
+          title, url = link_arguments(enrollment, text)
```
Comment (woodpigeon) [inline app/helpers/digital_services_core/back_link_helper.rb]: don't think you need to pass enrollment as its an attr_reader

File: app/helpers/digital_services_core/back_link_helper.rb [hunk 2]
```diff
@@ -4,27 +4,41 @@ module BackLinkHelper
...
+        end
       end
 
       private
 
       # Returns a [title, url] array for the current context
-      def link_arguments(enrollment)
-        if enrollment_is_mid_registration?
-          [I18n.translate("back"), enrollment_back_url]
+      def link_arguments(enrollment, text = nil)
```
Comment (woodpigeon) [inline app/helpers/digital_services_core/back_link_helper.rb]: don't think you need to pass enrollment as its an attr_reader

File: app/helpers/digital_services_core/back_link_helper.rb [hunk 3]
```diff
@@ -4,27 +4,41 @@ module BackLinkHelper
...
+
+        if enrollment.try! :under_review?
+          [text, enrollment_reviewing_path]
+        elsif enrollment_is_mid_registration?
+          [text, enrollment_back_url]
         else
-          [I18n.translate("backto_start_link"), start_url]
+          [text, start_url]
         end
       end
```
Comment (woodpigeon) [inline app/helpers/digital_services_core/back_link_helper.rb]: All good except method now > 10 lines :)
Here is an alternative sln keeping methods smaller - just an idea - could lose #link_arguments altogether and call #link_text(text) and #link_url from #tag
```
def link_arguments(text = nil)
  [text || link_text, link_url]
end

def link_text
  i18n_key = enrollment_is_mid_registration? ? "back" : "backto_start_link"
  I18n.translate i18n_key
end

def link_url
  if enrollment.try! :under_review?
    enrollment_reviewing_path
  elsif enrollment_is_mid_registration?
    enrollment_back_url
  else
    start_url
  end
end
```

**Comment by rickclare** [comment]
Thanks @woodpigeon all good comments - all implemented

**Pull request merged**